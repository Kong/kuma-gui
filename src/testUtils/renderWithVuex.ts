import { render, RenderOptions, RenderResult } from '@testing-library/vue'
import merge from 'lodash/merge'
import { StoreOptions } from 'vuex'
import Vue, { ComponentOptions } from 'vue'
import Store from '@/store'

function renderWithVuex(
  Component: typeof Vue | ComponentOptions<Vue>,
  options: RenderOptions<Vue> = { store: {}, routes: [] },
): RenderResult {
  const store: StoreOptions<any> = Store()

  // Render the component and merge the original store and the custom one
  // provided as a parameter. This way, we can alter some behaviors of the
  // initial implementation.

  return render(Component, { ...options, store: merge(store, options.store), routes: options.routes })
}

export default renderWithVuex
