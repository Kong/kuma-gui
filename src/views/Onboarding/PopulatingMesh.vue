<template>
  <OnboardingPage>
    <template #header>
      <OnboardingHeading title="3. Populating a Mesh" />
    </template>
    <template #content>
      <p class="text-center">
        {{ title }} is a multi-tenant system that can support multiple service meshes in the same cluster:
      </p>
      <div class="flex justify-center mt-10 mb-16 pb-16">
        <div class="w-full sm:w-3/5 lg:w-2/5 p-4">
          <KTable
            :options="tableData"
            is-small
          />
        </div>
      </div>
      <p class="text-center">
        {{ title }} automatically creates a “default” mesh that’s ready-to-use. Let’s add services next.
      </p>
    </template>

    <template #navigation>
      <OnboardingNavigation
        next-step="onboarding-adding-dpp"
        previous-step="onboarding-backend-types"
      />
    </template>
  </OnboardingPage>
</template>

<script>
import { mapGetters } from 'vuex'
import OnboardingNavigation from '@/views/Onboarding/components/OnboardingNavigation'
import OnboardingHeading from '@/views/Onboarding/components/OnboardingHeading'
import OnboardingPage from '@/views/Onboarding/components/OnboardingPage'

export default {
  name: 'PopulatingMesh',
  components: {
    OnboardingNavigation,
    OnboardingHeading,
    OnboardingPage,
  },
  metaInfo() {
    return {
      title: 'Populating a Mesh',
    }
  },
  data() {
    return {
      tableData: {
        headers: [
          { label: 'Name', key: 'name' },
          { label: 'Services', key: 'servicesAmount' },
          { label: 'DPPs', key: 'dppsAmount' },
        ],
        data: [{ name: 'default', servicesAmount: 0, dppsAmount: 0 }],
      },
    }
  },
  computed: {
    ...mapGetters({
      title: 'config/getTagline',
    }),
  },
}
</script>
