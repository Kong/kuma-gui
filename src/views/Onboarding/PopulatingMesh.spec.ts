import renderWithVuex from '@/testUtils/renderWithVuex'
import PopulatingMesh from './PopulatingMesh.vue'

describe('PopulatingMesh.vue', () => {
  const customStore = { modules: { config: { state: { tagline: 'Kuma' } } } }

  it('renders snapshot', async () => {
    const { container } = renderWithVuex(PopulatingMesh, { store: customStore })

    expect(container).toMatchSnapshot()
  })
})
