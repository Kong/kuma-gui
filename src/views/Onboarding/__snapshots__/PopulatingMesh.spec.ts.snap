// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopulatingMesh.vue renders snapshot 1`] = `
<div>
  <div>
    <div
      class="onboarding-container"
    >
      <div
        class="onboarding-container__header"
      >
        <div
          class="relative"
        >
          <h1
            class="onboarding-title"
          >
            
    Creating a mesh
  
          </h1>
           
          <p
            class="text-center text-lg mt-3"
          >
            
    Kuma is a multi-mesh service mesh, this means that we can create multiple different "Meshes" for our applications that are fully compartmentalized from each other to scale across an organization.
  
          </p>
        </div>
      </div>
       
      <div
        class="onboarding-container__content"
      >
        <div
          class="w-full"
        >
          <p
            class="text-center mb-4"
          >
            
      By default, Kuma creates a "default" mesh that we can start using for our services, but we can have
      as many meshes as we need:
    
          </p>
           
          <div
            class="flex justify-center mt-10 mb-12 pb-12"
          >
            <div
              class="w-full sm:w-3/5 lg:w-2/5 p-4"
            >
              <table
                class="k-table has-hover is-small side-border"
                data-v-bfeabd48=""
              >
                <thead
                  data-v-bfeabd48=""
                >
                  <tr
                    data-v-bfeabd48=""
                  >
                    <th
                      class=""
                      data-v-bfeabd48=""
                    >
                       Name 
                    </th>
                    <th
                      class=""
                      data-v-bfeabd48=""
                    >
                       Services 
                    </th>
                    <th
                      class=""
                      data-v-bfeabd48=""
                    >
                       DPPs 
                    </th>
                  </tr>
                </thead>
                <tbody
                  data-v-bfeabd48=""
                >
                  <tr
                    data-v-bfeabd48=""
                  >
                    <td
                      data-v-bfeabd48=""
                    >
                      default
                    </td>
                    <td
                      data-v-bfeabd48=""
                    >
                      0
                    </td>
                    <td
                      data-v-bfeabd48=""
                    >
                      0
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
           
          <p
            class="text-center"
          >
            
      The "default" mesh is currently empty with no services and therefore no data plane proxies. Let's go ahead and
      start a few services in the next step.
    
          </p>
        </div>
      </div>
       
      <div
        class="mt-4 flex items-center flex-col sm:flex-row justify-between"
      >
        <a
          aria-current="page"
          class="k-button navigation-button navigation-button--back router-link-exact-active router-link-active primary"
          data-v-38fa8ecf=""
          href="#/"
          type="button"
        >
          
    Back
  
        </a>
         
        <div>
          <button
            class="k-button skip-button small btn-link"
            data-v-38fa8ecf=""
            type="button"
          >
            
      Skip Setup
    
          </button>
           
          <a
            aria-current="page"
            class="k-button navigation-button navigation-button--next router-link-exact-active router-link-active primary"
            data-v-38fa8ecf=""
            href="#/"
            type="button"
          >
            
      Next
    
          </a>
        </div>
      </div>
    </div>
     
    <div
      class="background-image"
    />
  </div>
</div>
`;
